README
NAME
    Pod::Weaver::PluginBundle::Author::PERLANCAR - PERLANCAR's default
    Pod::Weaver config

VERSION
    This document describes version 0.19 of
    Pod::Weaver::PluginBundle::Author::PERLANCAR (from Perl distribution
    Pod-Weaver-PluginBundle-Author-PERLANCAR), released on 2014-11-23.

SYNOPSIS
    In "weaver.ini":

     [@Author::PERLANCAR]

    or in "dist.ini":

     [PodWeaver]
     config_plugin = @Author::PERLANCAR

DESCRIPTION
    Equivalent to (see source code).

SEE ALSO
    Pod::Weaver

    Dist::Zilla::PluginBundle::Author::PERLANCAR

HOMEPAGE
    Please visit the project's homepage at
    <https://metacpan.org/release/Pod-Weaver-PluginBundle-Author-PERLANCAR>.

SOURCE
    Source repository is at
    <https://github.com/perlancar/perl-Pod-Weaver-PluginBundle-Author-SHARYA
    NTO>.

BUGS
    Please report any bugs or feature requests on the bugtracker website
    <https://rt.cpan.org/Public/Dist/Display.html?Name=Pod-Weaver-PluginBund
    le-Author-PERLANCAR>

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR
    perlancar <perlancar@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by perlancar@cpan.org.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Pod-Weaver-PluginBundle-Author-PERLANCAR

0.19    2014-11-23 (PERLANCAR)

	- Update to use new Pod-Weaver-Section-Completion-* instead of the
	  old Pod-Weaver-Section-BashCompletion-* ones.


0.18    2014-11-21 (PERLANCAR)

        - Rename PWP::Perinci -> PWP::Rinci.


0.17    2014-11-12 (PERLANCAR)

	- Add section: BashCompletion::GetoptLongComplete.

        - Forgot to add dependencies.


0.16    2014-11-08 (PERLANCAR)

	- Add section: BashCompletion::PerinciCmdLine.


0.15    2014-08-26 (PERLANCAR)

        - Rename dist.


0.14    2014-04-29 (SHARYANTO)

        - Yet another wording change in VERSION section (remove 'module').


0.13    2014-04-29 (SHARYANTO)

        - Change wording in VERSION section ('distribution' -> 'Perl
          distribution').


0.12    2014-04-28 (SHARYANTO)

        - Remove PWS::ReleaseDate, PWS::Version is perfectly capable of
          displaying more than version number, including release date.


0.11    2014-04-23 (SHARYANTO)

        - Add plugin PWS::ReleaseDate.


0.10    2014-04-12 (SHARYANTO)

        - Add plugin bundle PWP::Author::SHARYANTO::NoPerinci (for lib-none,
          since the Perinci plugin will fail to load it; and there is currently
          no @Filter mechanism like in Dist::Zilla).


0.09    2013-11-12 (SHARYANTO)

        - Force version of Pod::Weaver::Section::Bugs::DefaultRT 0.03.


0.08    2013-11-12 (SHARYANTO)

        - Replace Encoding with SingleEncoding (thanks sergeyromanov). Require
          Pod::Weaver 4.0.


0.07    2013-11-08 (SHARYANTO)

        - No functional changes. Force plugin versions.


0.06    2013-10-31 (SHARYANTO)

        - Add missing section: Version.


0.05    2013-10-22 (SHARYANTO)

        - Make this PW bundle usable for CPAN modules as well as non-CPAN
          modules.

        - Replace Availability with Homepage::DefaultCPAN.

        - Replace SourceGitHub with Source::DefaultGitHub.

        - Replace BugsRT with Bugs::DefaultRT.


0.04    2013-10-19 (SHARYANTO)

        - Rename from Pod-Weaver-PluginBundle-SHARYANTO to
          Pod-Weaver-PluginBundle-Author-SHARYANTO.

        - Add sections: Availability & SourceGitHub (because people reading the
          POD and not sco/metacpan wouldn't realize that the module is hosted on
          GitHub).

        - Add section: BugsRT.

        - Fix ordering.


0.03    2013-04-10 (SHARYANTO)

        - Add Encoding.


0.02    2013-04-04 (SHARYANTO)

        - No functional changes. Mark with /PW so that the log message source is
          clear when doing e.g. 'dzil build -v'.


0.01    2013-04-04 (SHARYANTO)

        - First release.


lib/Pod/Weaver/PluginBundle/Author/PERLANCAR.pm
package Pod::Weaver::PluginBundle::Author::PERLANCAR;

our $DATE = '2014-11-23'; # DATE
our $VERSION = '0.19'; # VERSION

use 5.010001;

use Pod::Weaver::Config::Assembler;

sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub mvp_bundle_config {
    return (
        [ '@Author::PERLANCAR/CorePrep', _exp('@CorePrep'), {} ],
        [ '@Author::PERLANCAR/Name', _exp('Name'), {} ],
        [ '@Author::PERLANCAR/Version', _exp('Version'), {format=>'This document describes version %v of %m (from Perl distribution %r), released on %{YYYY-MM-dd}d.'} ],
        [ '@Author::PERLANCAR/prelude', _exp('Region'), { region_name => 'prelude' } ],

        [ 'SYNOPSIS', _exp('Generic'), {} ],
        [ 'DESCRIPTION', _exp('Generic'), {} ],
        [ 'OVERVIEW', _exp('Generic'), {} ],

        [ 'ATTRIBUTES', _exp('Collect'), { command => 'attr' } ],
        [ 'METHODS', _exp('Collect'), { command => 'method' } ],
        [ 'FUNCTIONS', _exp('Collect'), { command => 'func' } ],
        #[ 'TYPES', _exp('Collect'), { command => ' } ],

        [ '@Author::PERLANCAR/Rinci', _exp('-Rinci'), {} ],

        [ '@Author::PERLANCAR/Leftovers', _exp('Leftovers'), {} ],
        [ '@Author::PERLANCAR/postlude',  _exp('Region'), { region_name => 'postlude' } ],

        [ '@Author::PERLANCAR/Completion::GetoptLongComplete', _exp('Completion::GetoptLongComplete'), {} ],
        [ '@Author::PERLANCAR/Completion::PerinciCmdLine', _exp('Completion::PerinciCmdLine'), {} ],

        [ '@Author::PERLANCAR/Homepage::DefaultCPAN', _exp('Homepage::DefaultCPAN'), {} ],
        [ '@Author::PERLANCAR/Source::DefaultGitHub', _exp('Source::DefaultGitHub'), {} ],
        [ '@Author::PERLANCAR/Bugs::DefaultRT', _exp('Bugs::DefaultRT'), {} ],
        [ '@Author::PERLANCAR/Authors', _exp('Authors'), {} ],
        [ '@Author::PERLANCAR/Legal', _exp('Legal'), {} ],

        [ '@Author::PERLANCAR/EnsureUniqueSections', _exp('-EnsureUniqueSections'), {} ],
        #[ '@Author::PERLANCAR/List', _exp('-Transformer'), { transformer => 'List' } ],
        [ '@Author::PERLANCAR/SingleEncoding', _exp('-SingleEncoding'), {} ],
    );
}

1;
# ABSTRACT: PERLANCAR's default Pod::Weaver config

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::Author::PERLANCAR - PERLANCAR's default Pod::Weaver config

=head1 VERSION

This document describes version 0.19 of Pod::Weaver::PluginBundle::Author::PERLANCAR (from Perl distribution Pod-Weaver-PluginBundle-Author-PERLANCAR), released on 2014-11-23.

=head1 SYNOPSIS

In C<weaver.ini>:

 [@Author::PERLANCAR]

or in C<dist.ini>:

 [PodWeaver]
 config_plugin = @Author::PERLANCAR

=head1 DESCRIPTION

Equivalent to (see source code).

=for Pod::Coverage .*

=head1 SEE ALSO

L<Pod::Weaver>

L<Dist::Zilla::PluginBundle::Author::PERLANCAR>

=head1 HOMEPAGE

Please visit the project's homepage at L<https://metacpan.org/release/Pod-Weaver-PluginBundle-Author-PERLANCAR>.

=head1 SOURCE

Source repository is at L<https://github.com/perlancar/perl-Pod-Weaver-PluginBundle-Author-SHARYANTO>.

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website L<https://rt.cpan.org/Public/Dist/Display.html?Name=Pod-Weaver-PluginBundle-Author-PERLANCAR>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

perlancar <perlancar@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by perlancar@cpan.org.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/Pod/Weaver/PluginBundle/Author/PERLANCAR/NoRinci.pm
package Pod::Weaver::PluginBundle::Author::PERLANCAR::NoRinci;

our $DATE = '2014-11-23'; # DATE
our $VERSION = '0.19'; # VERSION

use 5.010001;

use Pod::Weaver::Config::Assembler;

sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub mvp_bundle_config {
    return (
        [ '@Author::PERLANCAR/CorePrep', _exp('@CorePrep'), {} ],
        [ '@Author::PERLANCAR/Name', _exp('Name'), {} ],
        [ '@Author::PERLANCAR/Version', _exp('Version'), {format=>'This document describes version %v of %m (from Perl distribution %r), released on %{YYYY-MM-dd}d.'} ],
        [ '@Author::PERLANCAR/prelude', _exp('Region'), { region_name => 'prelude' } ],

        [ 'SYNOPSIS', _exp('Generic'), {} ],
        [ 'DESCRIPTION', _exp('Generic'), {} ],
        [ 'OVERVIEW', _exp('Generic'), {} ],

        [ 'ATTRIBUTES', _exp('Collect'), { command => 'attr' } ],
        [ 'METHODS', _exp('Collect'), { command => 'method' } ],
        [ 'FUNCTIONS', _exp('Collect'), { command => 'func' } ],
        #[ 'TYPES', _exp('Collect'), { command => ' } ],

        [ '@Author::PERLANCAR/Leftovers', _exp('Leftovers'), {} ],
        [ '@Author::PERLANCAR/postlude',  _exp('Region'), { region_name => 'postlude' } ],

        [ '@Author::PERLANCAR/Completion::GetoptLongComplete', _exp('Completion::GetoptLongComplete'), {} ],
        [ '@Author::PERLANCAR/Completion::PerinciCmdLine', _exp('Completion::PerinciCmdLine'), {} ],

        [ '@Author::PERLANCAR/Homepage::DefaultCPAN', _exp('Homepage::DefaultCPAN'), {} ],
        [ '@Author::PERLANCAR/Source::DefaultGitHub', _exp('Source::DefaultGitHub'), {} ],
        [ '@Author::PERLANCAR/Bugs::DefaultRT', _exp('Bugs::DefaultRT'), {} ],
        [ '@Author::PERLANCAR/Authors', _exp('Authors'), {} ],
        [ '@Author::PERLANCAR/Legal', _exp('Legal'), {} ],

        [ '@Author::PERLANCAR/EnsureUniqueSections', _exp('-EnsureUniqueSections'), {} ],
        #[ '@Author::PERLANCAR/List', _exp('-Transformer'), { transformer => 'List' } ],
        [ '@Author::PERLANCAR/SingleEncoding', _exp('-SingleEncoding'), {} ],
    );
}

1;
# ABSTRACT: PERLANCAR's default Pod::Weaver config

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::Author::PERLANCAR::NoRinci - PERLANCAR's default Pod::Weaver config

=head1 VERSION

This document describes version 0.19 of Pod::Weaver::PluginBundle::Author::PERLANCAR::NoRinci (from Perl distribution Pod-Weaver-PluginBundle-Author-PERLANCAR), released on 2014-11-23.

=head1 SYNOPSIS

In C<weaver.ini>:

 [@Author::PERLANCAR]

or in C<dist.ini>:

 [PodWeaver]
 config_plugin = @Author::PERLANCAR

=head1 DESCRIPTION

Equivalent to (see source code).

=for Pod::Coverage .*

=head1 SEE ALSO

L<Pod::Weaver>

L<Dist::Zilla::PluginBundle::Author::PERLANCAR>

=head1 HOMEPAGE

Please visit the project's homepage at L<https://metacpan.org/release/Pod-Weaver-PluginBundle-Author-PERLANCAR>.

=head1 SOURCE

Source repository is at L<https://github.com/perlancar/perl-Pod-Weaver-PluginBundle-Author-SHARYANTO>.

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website L<https://rt.cpan.org/Public/Dist/Display.html?Name=Pod-Weaver-PluginBundle-Author-PERLANCAR>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

perlancar <perlancar@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by perlancar@cpan.org.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/Pod/Weaver/PluginBundle/Author/PERLANCAR/NoPerinci.pm
package Pod::Weaver::PluginBundle::Author::PERLANCAR::NoPerinci;

our $DATE = '2014-11-12'; # DATE
our $VERSION = '0.17'; # VERSION

use 5.010001;

use Pod::Weaver::Config::Assembler;

sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub mvp_bundle_config {
    return (
        [ '@Author::PERLANCAR/CorePrep', _exp('@CorePrep'), {} ],
        [ '@Author::PERLANCAR/Name', _exp('Name'), {} ],
        [ '@Author::PERLANCAR/Version', _exp('Version'), {format=>'This document describes version %v of %m (from Perl distribution %r), released on %{YYYY-MM-dd}d.'} ],
        [ '@Author::PERLANCAR/prelude', _exp('Region'), { region_name => 'prelude' } ],

        [ 'SYNOPSIS', _exp('Generic'), {} ],
        [ 'DESCRIPTION', _exp('Generic'), {} ],
        [ 'OVERVIEW', _exp('Generic'), {} ],

        [ 'ATTRIBUTES', _exp('Collect'), { command => 'attr' } ],
        [ 'METHODS', _exp('Collect'), { command => 'method' } ],
        [ 'FUNCTIONS', _exp('Collect'), { command => 'func' } ],
        #[ 'TYPES', _exp('Collect'), { command => ' } ],

        [ '@Author::PERLANCAR/Leftovers', _exp('Leftovers'), {} ],
        [ '@Author::PERLANCAR/postlude',  _exp('Region'), { region_name => 'postlude' } ],

        [ '@Author::PERLANCAR/BashCompletion::GetoptLongComplete', _exp('BashCompletion::GetoptLongComplete'), {} ],
        [ '@Author::PERLANCAR/BashCompletion::PerinciCmdLine', _exp('BashCompletion::PerinciCmdLine'), {} ],

        [ '@Author::PERLANCAR/Homepage::DefaultCPAN', _exp('Homepage::DefaultCPAN'), {} ],
        [ '@Author::PERLANCAR/Source::DefaultGitHub', _exp('Source::DefaultGitHub'), {} ],
        [ '@Author::PERLANCAR/Bugs::DefaultRT', _exp('Bugs::DefaultRT'), {} ],
        [ '@Author::PERLANCAR/Authors', _exp('Authors'), {} ],
        [ '@Author::PERLANCAR/Legal', _exp('Legal'), {} ],

        [ '@Author::PERLANCAR/EnsureUniqueSections', _exp('-EnsureUniqueSections'), {} ],
        #[ '@Author::PERLANCAR/List', _exp('-Transformer'), { transformer => 'List' } ],
        [ '@Author::PERLANCAR/SingleEncoding', _exp('-SingleEncoding'), {} ],
    );
}

1;
# ABSTRACT: PERLANCAR's default Pod::Weaver config

__END__

=pod

=encoding UTF-8

=head1 NAME

Pod::Weaver::PluginBundle::Author::PERLANCAR::NoPerinci - PERLANCAR's default Pod::Weaver config

=head1 VERSION

This document describes version 0.17 of Pod::Weaver::PluginBundle::Author::PERLANCAR::NoPerinci (from Perl distribution Pod-Weaver-PluginBundle-Author-PERLANCAR), released on 2014-11-12.

=head1 SYNOPSIS

In C<weaver.ini>:

 [@Author::PERLANCAR]

or in C<dist.ini>:

 [PodWeaver]
 config_plugin = @Author::PERLANCAR

=head1 DESCRIPTION

Equivalent to (see source code).

=for Pod::Coverage .*

=head1 SEE ALSO

L<Pod::Weaver>

L<Dist::Zilla::PluginBundle::Author::PERLANCAR>

=head1 HOMEPAGE

Please visit the project's homepage at L<https://metacpan.org/release/Pod-Weaver-PluginBundle-Author-PERLANCAR>.

=head1 SOURCE

Source repository is at L<https://github.com/perlancar/perl-Pod-Weaver-PluginBundle-Author-SHARYANTO>.

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website L<https://rt.cpan.org/Public/Dist/Display.html?Name=Pod-Weaver-PluginBundle-Author-PERLANCAR>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

perlancar <perlancar@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by perlancar@cpan.org.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.019.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "PERLANCAR's default Pod::Weaver config",
  "AUTHOR" => "perlancar <perlancar\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Pod-Weaver-PluginBundle-Author-PERLANCAR",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Pod::Weaver::PluginBundle::Author::PERLANCAR",
  "PREREQ_PM" => {
    "Pod::Weaver" => "4.000",
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::EnsureUniqueSections" => 0,
    "Pod::Weaver::Plugin::Rinci" => 0,
    "Pod::Weaver::Section::Bugs::DefaultRT" => "0.03",
    "Pod::Weaver::Section::Completion::GetoptLongComplete" => "0.02",
    "Pod::Weaver::Section::Completion::PerinciCmdLine" => "0.04",
    "Pod::Weaver::Section::Homepage::DefaultCPAN" => 0,
    "Pod::Weaver::Section::Source::DefaultGitHub" => "0.04"
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.19",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Pod::Weaver" => "4.000",
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::EnsureUniqueSections" => 0,
  "Pod::Weaver::Plugin::Rinci" => 0,
  "Pod::Weaver::Section::Bugs::DefaultRT" => "0.03",
  "Pod::Weaver::Section::Completion::GetoptLongComplete" => "0.02",
  "Pod::Weaver::Section::Completion::PerinciCmdLine" => "0.04",
  "Pod::Weaver::Section::Homepage::DefaultCPAN" => 0,
  "Pod::Weaver::Section::Source::DefaultGitHub" => "0.04",
  "Test::More" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



